src/main/java/org/thingsboard/gateway/service/gateway/MqttGatewayClient.java

```diff
@@ -0,0 +1,580 @@
+                });
+    }
+
+    public void sendDeviceRpcResponse(RpcCommandResponse response) {
+        final int msgId = msgIdSeq.incrementAndGet();
+        int requestId = response.getRequestId();
+        String deviceName = response.getDeviceName();
+        String data = response.getData();
+
+        ObjectNode node = newNode();
+        node.put("id", requestId);
+        node.put("device", deviceName);
+        node.put("data", fromString(data));
+        persistMessage(GATEWAY_RPC_TOPIC, msgId, toBytes(node), deviceName,
+                token -> {
+                    log.debug("[{}][{}] RPC response from device was delivered!", deviceName, requestId);
+                },
+                error -> {
+                    log.warn("[{}][{}] Failed to report RPC response from device!", deviceName, requestId, error);
+                });
+    }
+
+    public void onError(Exception e) {
+        onError(null, e);
+    }
+
+    public void onError(String deviceName, Exception e) {
+        ObjectNode node = newNode();
+        node.put("ts", System.currentTimeMillis());
+        if (deviceName != null) {
+            node.put("device", deviceName);
+        }
+        node.put("error", toString(e));
+        error = node;
+    }
+
+    public void sendStatsReport(ObjectNode values) {
+        try {
+            ObjectNode node = newNode();
+            node.put("ts", System.currentTimeMillis());
+            ObjectNode valuesNode = node.putObject("values");
+
+            valuesNode.setAll(values);
+            valuesNode.put("attributesUploaded", attributesCount.getAndSet(0));
+            valuesNode.put("telemetryUploaded", telemetryCount.getAndSet(0));
+            if (error != null) {
+                valuesNode.put("latestError", JsonTools.toString(error));
+                error = null;
+            }
+            persistMessage(DEVICE_TELEMETRY_TOPIC, msgIdSeq.incrementAndGet(), toBytes(node), GATEWAY,
+                    token -> log.info("Gateway statistics {} reported!", node),
+                    error -> log.warn("Failed to report gateway statistics!", error));
+        } catch (Throwable e) {
+            log.error("Failed to persist statistics message!", e);
+        }
+    }
+
+    @Override
+    public void onMessage(String topic, ByteBuf payload) {
+        String message = payload.toString(StandardCharsets.UTF_8);
+
+        log.trace("Message arrived [{}] {}", topic, message);
+        callbackExecutor.submit(() -> {
+            try {
+                if (topic.equals(GATEWAY_ATTRIBUTES_TOPIC)) {
+                    onAttributesUpdate(message);
+                } else if (topic.equals(GATEWAY_RESPONSES_ATTRIBUTES_TOPIC)) {
+                    onDeviceAttributesResponse(message);
+                } else if (topic.equals(GATEWAY_RPC_TOPIC)) {
+                    onRpcCommand(message);
+                } else if (topic.startsWith(DEVICE_RPC_REQUEST_TOPIC)) {
+                    // TODO: add RPC command process
```

Comment:
Please add processing for DEVICE_RPC_REQUEST_TOPIC, DEVICE_RPC_REQUEST_PLUS_TOPIC , DEVICE_RPC_RESPONSE_TOPIC